spec/workflow/intrinsic_function_spec.rb

```diff
@@ -610,6 +617,13 @@
       end
     end
 
+    describe "States.StringToJson" do
+      it "fetches values from the input" do
+        result = described_class.value("States.StringToJson($.input)", {"context" => {"baz" => "qux"}}, {"input" => "{\"foo\":\"bar\"}"})
+        expect(result).to eq({"foo" => "bar"})
+      end
```

Comment:
Please also add the common failure cases as before.

Addtionally, there can be "bad" strings, so let's have specs for those as well (and on that note, they will probably raise a JSON::ParserError that we probably need to capture and convert to a Floe error.  Offhand, this should non-json strings (e.g. "XXX"), the empty string, a string with only a newline, and non-hash JSON strings like Array, Integer, "null", etc (there's probably more we can verify).